pennylane/ops/functions/equal.py

```diff
@@ -151,7 +160,7 @@ def equal(
     if not isinstance(op2, type(op1)) and not isinstance(op1, Observable):
         return False
 
-    if isinstance(op2, (Hamiltonian, Tensor)):
+    if isinstance(op2, (Hamiltonian, LinearCombination, Tensor)):
```

Comment:
```suggestion
    if isinstance(op2, (Hamiltonian, Tensor)):
```
I would prefer to follow the more generic `CompositeOp` approach instead and be able to use all the keyword args. 

We should have it that:
```
>>> op1 = qml.LinearCombination([1, 2], [qml.X(0), qml.X(1)])
>>> op2 = qml.LinearCombination([1+1e-5, 2], [qml.X(0), qml.X(1)])
>>> qml.equal(op1, op2, atol=1e-3)
True
>>> qml.equal(op1, op2, atol=1e-6)
False
>>> op3 = qml.LinearCombination([qml.numpy.array(1), 2], [qml.X(0), qml.X(1)])
>>> qml.equal(op3, op1, check_interface=True)
False
```

Would be willing to break these changes into another task if necesssary.

Reply:
Added to https://app.shortcut.com/xanaduai/story/59353/upgrade-linearcombination and marking as resolved here